File: src/main/java/nz/co/searchwellington/model/NewsitemImpl.java
package nz.co.searchwellington.model;

import java.util.Collections;
import java.util.Date;
import java.util.List;
import java.util.Set;

import com.google.common.collect.ImmutableList;

public class NewsitemImpl extends PublishedResourceImpl implements Newsitem {
	
	private static final long serialVersionUID = 1L;
	
	CommentFeed commentFeed;	// TODO should be visible on the interface
    Image image;
    Feed feed;
    Date accepted;
    User acceptedBy;

    public NewsitemImpl() {
    }
    
    public NewsitemImpl(int id, String name, String url, String description, Date date, Website publisher, Set<DiscoveredFeed> discoveredFeeds) {
        this.id = id;
        this.name = name;
        this.url = url;
        this.description = description;
        this.date = date;
        this.publisher = publisher;
        this.discoveredFeeds = discoveredFeeds;
        this.feed = null;
    }
    
    public String getType() {
        return "N";
    }
    
    public List<Comment> getComments() {
    	if (getCommentFeed() != null) {
    		return ImmutableList.<Comment>builder().addAll(getCommentFeed().getComments()).build();
    	}
    	return Collections.emptyList();
	}

	public CommentFeed getCommentFeed() {
        return commentFeed;
    }

    public void setCommentFeed(CommentFeed commentFeed) {
        this.commentFeed = commentFeed;
    }
    
	public Image getImage() {
		return image;
	}
	
	public void setImage(Image image) {
		this.image = image;
	}

	public Feed getFeed() {
		return feed;
	}

	public void setFeed(Feed feed) {
		this.feed = feed;
	}

	public Date getAccepted() {
		return accepted;
	}

	public void setAccepted(Date accepted) {
		this.accepted = accepted;
	}

	public User getAcceptedBy() {
		return acceptedBy;
	}

	public void setAcceptedBy(User acceptedBy) {
		this.acceptedBy = acceptedBy;
	}
	
}

File: src/main/scala/nz/co/searchwellington/model/NewsitemImpl.scala
package nz.co.searchwellington.model

import java.util.Date

import org.joda.time.format.ISODateTimeFormat

case class NewsitemImpl(override var id: Int = 0,
                        override var `type`: String = "",
                        override var title: Option[String] = None,
                        override var page: Option[String] = None,
                        override var http_status: Int = 0,
                        override var date2: Option[Date] = None,
                        override var description: Option[String] = None,
                        override var last_scanned2: Option[Date] = None,
                        override var last_changed2: Option[Date] = None,
                        override var live_time2: Option[Date] = None,
                        override var embargoed_until2: Option[Date] = None,
                        override var held2: Boolean = true,
                        override var url_words: Option[String] = None,
                        override var geocode: Option[Int] = None,
                        override var owner: Option[Int] = None,
                        override var publisher: Option[Int] = None,
                        var feed: Option[Int] = None,
                        var commentFeed: Option[Int] = None,
                        var image: Option[Int] = None,
                        var accepted: Option[String] = None,
                        var acceptedBy: Option[Int] = None) extends PublishedResource with Newsitem {

  override def getType = "N"

  //override def getComments: Seq[Comment] = {
    //if (getCommentFeed != null) return ImmutableList.builder[Comment].addAll(getCommentFeed.getComments).build
    //Collections.emptyList
   // Seq()
 // }

  def getCommentFeed: Option[Int] = commentFeed

  def setCommentFeed(commentFeed: Int): Unit = this.commentFeed = Some(commentFeed)

  override def getImage: Option[Int] = image

  override def setImage(image: Int): Unit = this.image = Some(image)

  override def getFeed: Option[Int] = feed

  override def setFeed(feed: Int): Unit = this.feed = Some(feed)

  override def getAccepted: Date = accepted.map(d => ISODateTimeFormat.dateParser().parseDateTime(d).toDate).getOrElse(null) // TODO

  override def setAccepted(accepted: Date): Unit = this.accepted = Some(accepted.toString)  // TODO

  override def getAcceptedBy: Option[Int] = acceptedBy

  override def setAcceptedBy(acceptedBy: Int): Unit = this.acceptedBy = Some(acceptedBy)

}
